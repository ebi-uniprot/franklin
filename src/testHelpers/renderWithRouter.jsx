import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

export default (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  };
};
