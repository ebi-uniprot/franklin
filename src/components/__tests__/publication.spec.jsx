import React from 'react';
import { fireEvent } from '@testing-library/react';
import Publication from '../publication';
import publicationData from '../__mocks__/publications';
import renderWithRouter from '../../testHelpers/renderWithRouter';

let rendered;

describe('Publication component', () => {
  beforeEach(() => {
    const {
      title,
      authors,
      abstract,
      infoData,
      journalInfo,
      pubmedId,
      statistics,
    } = publicationData;

    rendered = renderWithRouter(
      <Publication
        title={title}
        authors={authors}
        abstract={abstract}
        infoData={infoData}
        journalInfo={journalInfo}
        pubmedId={pubmedId}
        statistics={statistics}
      />
    );
  });

  test('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should expand authors', async () => {
    const { queryByText, getByText, findByText } = rendered;
    expect(queryByText('Darwin K.')).toBeNull();
    fireEvent.click(getByText('[...]'));
    const author = await findByText('Darwin K.');
    expect(author).toBeTruthy();
  });

  test('should expand abstract', async () => {
    const { queryByText, getByText, findByText } = rendered;
    expect(queryByText(/Captain Fitzroy/)).toBeNull();
    fireEvent.click(getByText('View abstract [...]'));
    const abstract = await findByText(/Captain Fitzroy/);
    expect(abstract).toBeTruthy();
  });
});
