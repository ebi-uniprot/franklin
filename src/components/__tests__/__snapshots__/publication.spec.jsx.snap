// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publication component should render 1`] = `
<DocumentFragment>
  <section
    class="publication"
  >
    <section
      class="publication__columns"
    >
      <section
        class="publication__columns__main"
      >
        <h5>
          The Zoology of the Voyage of H.M.S. Beagle.
        </h5>
        <section
          class="publication__authors"
        >
          <a
            href="/"
          >
            Darwin C.
          </a>
          , 
          <a
            href="/"
          >
            Darwin B.
          </a>
          , 
          <a
            href="/"
          >
            Darwin A.
          </a>
          , 
          <a
            href="/"
          >
            Darwin D.
          </a>
          , 
          <a
            href="/"
          >
            Darwin E.
          </a>
          , 
          <a
            href="/"
          >
            Darwin F.
          </a>
          , 
          <a
            href="/"
          >
            Darwin G.
          </a>
          , 
          <a
            href="/"
          >
            Darwin H.
          </a>
          , 
          <a
            href="/"
          >
            Darwin I.
          </a>
          <button
            class="button tertiary"
            type="button"
          >
            [...]
          </button>
          , 
          <a
            href="/"
          >
            Darwin X.
          </a>
        </section>
        <section
          class="publication__abstract"
        >
          <button
            class="button tertiary"
            type="button"
          >
            View abstract [...]
          </button>
        </section>
        <div
          class="info-list info-list--compact"
        >
          <div
            class="info-list__item"
          >
            <div
              class="info-list__title"
            >
              <h5
                class="bold"
              >
                Tortoise
              </h5>
            </div>
            <div
              class="info-list__content"
            >
              <em>
                Reptile species of the family Testudinidae of the order Testudines (the turtles).
              </em>
            </div>
          </div>
          <div
            class="info-list__item"
          >
            <div
              class="info-list__title"
            >
              <h5
                class="bold"
              >
                Beetles
              </h5>
            </div>
            <div
              class="info-list__content"
            >
              Charles really liked them.
            </div>
          </div>
        </div>
      </section>
      <section
        class="publication__columns__side"
      >
        <section
          class="publication__columns__side__item"
        >
          <test-file-stub
            height="2em"
            width="1.875em"
          />
          <ul
            class="no-bullet"
          >
            <li>
              <a
                href="//www.ncbi.nlm.nih.gov/pubmed/123456"
                rel="noopener noreferrer"
                target="_blank"
              >
                PubMed
              </a>
            </li>
            <li>
              <a
                href="//europepmc.org/article/MED/123456"
                rel="noopener noreferrer"
                target="_blank"
              >
                Europe PMC
              </a>
            </li>
            <li>
              <small>
                <a
                  href="//dx.doi.org/10.1038/331525a0"
                >
                  Nature 331:525-527(1988)
                </a>
              </small>
            </li>
          </ul>
        </section>
        <section
          class="publication__columns__side__item"
        >
          <section
            class="publication__statistics"
          >
            <section
              class="publication__statistics__item"
            >
              <a
                href="/"
              >
                <section>
                  <small>
                    Mapped to
                  </small>
                </section>
                <section
                  class="publication__statistics__bubble"
                >
                  <span
                    class="bubble--small colour-pastel-blue"
                  >
                    3
                  </span>
                  <test-file-stub
                    height="15"
                    width="15"
                  />
                </section>
              </a>
            </section>
            <section
              class="publication__statistics__item"
            >
              <a
                href="/"
              >
                <section>
                  <small>
                    Cited in
                  </small>
                </section>
                <section
                  class="publication__statistics__bubble"
                >
                  <span
                    class="bubble--small colour-pastel-blue"
                  >
                    1
                  </span>
                  <test-file-stub
                    height="15"
                    width="15"
                  />
                </section>
              </a>
            </section>
          </section>
        </section>
      </section>
    </section>
  </section>
</DocumentFragment>
`;
